docs/service_architecture.md

```diff
@@ -166,6 +169,87 @@ keys in the HSM, but not to exfiltrate the private material.
   signing arbitrary metadata, with the important detail that in this particular
   case key rotations will be necessary to recover from the attack.
 
+### Notary client keys and credentials compromise
+
+In the event of a trust collection owner/administrator's keys being compromised,
+an attacker would be able to sign valid changes to the contents of that
```

Comment:
It might be useful here to preface the subsections by saying the severity of the compromise depends on which kind of key in the hierarchy is compromised, as well as if it was a combination of keys (ex: snapshot key alone vs. snapshot + targets keys)